### PR TITLE
Webcast 2.58 (DVR + polls)

### PR DESCRIPTION
**Pull request by eitanavgil**

### CODE REVIEW

File: modules/KalturaSupport/resources/webcast/cuePointsManager.js [hunk 1]
```diff
@@ -149,6 +162,9 @@
                 _this.resetMonitorVariables();
             }
         },
+		startMonitorProcessPush : function(){
```
Comment (kalturaguy) [inline modules/KalturaSupport/resources/webcast/cuePointsManager.js]: ??

File: modules/KalturaSupport/resources/webcast/cuePointsManager.js [hunk 2]
```diff
@@ -300,33 +316,72 @@
...
+						mw.log("KPushCuePointsManager cuePoints loaded " + cuePoints);
+						_this.handleMonitoredCuepoints(cuePoints);
+						// _this.cuepointLoaded(cuePoints,cuepointTypes,callback,systemNames[i])
+					});
+				tempNotifications.push(tempNotification);
+			}
+			return this.pushServerNotification.registerNotifications(tempNotifications,pluginName);
+		},
+        //Proxy function
+        cuepointLoaded : function (cuePoints,cuepointTypes,callback,systemName){
```
Comment (kalturaguy) [inline modules/KalturaSupport/resources/webcast/cuePointsManager.js]: not used

File: modules/KalturaSupport/resources/webcast/cuePointsManager.js [hunk 3]
```diff
@@ -476,10 +530,17 @@
 
             return result;
         },
+        getCurrentTime: function(){
+            if(this.embedPlayer.isDVR()){
```
Comment (kalturaguy) [inline modules/KalturaSupport/resources/webcast/cuePointsManager.js]: isLive and not isDVR

File: modules/WebcastPolls/resources/webcastPollsKalturaProxy.js [hunk 4]
```diff
@@ -126,84 +126,142 @@
 
             return defer.promise();
         },
-        transmitVoteUpdate : function(metadataId, userId, selectedAnswer,pollId)
+        transmitVoteUpdate : function(metadataId, userId, selectedAnswer,pollId ,newApi)
```
Comment (kalturaguy) [inline modules/WebcastPolls/resources/webcastPollsKalturaProxy.js]: @eitanavgil 
a better approcach is to create a different class called webcastPollsKalturaProxy.js and webcastPollsKalturaProxyLegacy.js

and instasiate either this or that according to the newAPI configururation

Comment (eitanavgil) [inline modules/WebcastPolls/resources/webcastPollsKalturaProxy.js]: @kalturaguy  - is there a point of leaving the old functionality? As I see it - BE supports both, versions > V2.58 will work with new API, <=2.57 with old API 
I did this new flag so our QA will be able to test the both DVR and new API without conflicts. 
Do you see a reason to keep legacy too?

**Comment by eitanavgil** [comment]
Moved to 
https://github.com/kaltura/mwEmbed/tree/webcast_2.58_post_dvr 
https://github.com/kaltura/mwEmbed/pull/3424/files

**Pull request closed without merging**